lib/miq_pglogical.rb

```diff
@@ -0,0 +1,87 @@
+class MiqPglogical
+  REPLICATION_SET_NAME = 'miq'.freeze
+  SETTINGS_PATH = [:replication].freeze
+  NODE_PREFIX = "region_".freeze
+
+  def initialize(connection = ApplicationRecord.connection)
+    @connection = connection
+  end
+
+  # Configures the current database server to match the miq server configuration
+  def configure
+    mode = MiqServer.my_server.get_config.config.fetch_path(*SETTINGS_PATH, :mode)
+    case mode
+    when "local"
+      provider = MiqPglogicalProvider.new(@connection)
+      provider.configure_provider
+      provider.refresh_excludes
+    when "global"
+      subscriber = MiqPglogicalSubscriber.new(@connection)
+      subscriber.configure_subscriber
+      subscriber.refresh_subscriptions
```

Comment:
It would be cool if these two interfaces were similar and didn't need the subclass specific naming.